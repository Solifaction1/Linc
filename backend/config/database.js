const path = require('path');

const connection = process.env.NODE_ENV !== 'production' ?
  {
    client: 'sqlite',
    connection: {
      filename: path.join(__dirname, '..', '.tmp/data.db'),
    },
    useNullAsDefault: true,
  } : {
    client: "postgres",
    acquireConnectionTimeout: 120000,
    connection: {
      // host: `/cloudsql/${process.env.INSTANCE_CONNECTION_NAME}`, // for a PostgreSQL database
      host: process.env.DATABASE_HOST, // for a PostgreSQL database
      database: process.env.DATABASE_NAME,
      user: process.env.DATABASE_USERNAME,
      password: process.env.DATABASE_PASSWORD,
      ssl: false,

    },
    useNullAsDefault: true,
    debug: false,
  };

module.exports = ({ }) => ({
  connection: connection,
});
