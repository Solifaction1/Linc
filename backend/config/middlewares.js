module.exports = [
  'strapi::errors',
  {
    name: 'strapi::cors',
    config: {
      enabled: true,
      headers: ['*'],
      origin: ['*']
    }
  },
  'strapi::poweredBy',
  'strapi::logger',
  'strapi::query',
  'strapi::body',
  'strapi::session',
  'strapi::favicon',
  'strapi::public',
  {
    name: 'strapi::security',
    config: {
      contentSecurityPolicy: {
        useDefaults: true,
        directives: {
          'connect-src': ["'self'", 'https:'],
          'img-src': [
            "'self'",
            'data:',
            'blob:',
            'dl.airtable.com',
            "cdn.jsdelivr.net",
            "strapi.io",
            "s3.amazonaws.com",
            "editor.unlayer.com",
            process.env.CDN_URL,
          ],
          "script-src": ["'self'", "editor.unlayer.com"],
          "frame-src": ["'self'", "editor.unlayer.com"],
          'media-src': [
            "'self'",
            'data:',
            'blob:',
            'dl.airtable.com',
            process.env.CDN_URL,
          ],
          upgradeInsecureRequests: null,
        },
      },
    },
  },
];
