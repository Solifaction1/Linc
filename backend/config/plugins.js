module.exports = ({ env }) => ({
  "strapi-plugin-populate-deep": {
    config: {
      defaultDepth: 5, // Default is 5
    },
  },
  "duplicate-button": true,
  email: {
    config: {
      provider: "mailgun",
      providerOptions: {
        key: env("MAILGUN_API_KEY"), // Required
        domain: env("MAILGUN_DOMAIN"), // Required
        url: "https://api.eu.mailgun.net", //Optional. If domain region is Europe use 'https://api.eu.mailgun.net'
      },
      settings: {
        defaultFrom: "<mg@linc.legal>",
        defaultReplyTo: "<mg@linc.legal>",
      },
    },
  },
  upload: {
    config: {
      provider: "aws-s3",
      providerOptions: {
        accessKeyId: env("AWS_ACCESS_KEY_ID"),
        secretAccessKey: env("AWS_ACCESS_SECRET"),
        region: env("AWS_REGION"),
        params: {
          Bucket: env("AWS_BUCKET"),
        },
      },
      actionOptions: {
        upload: {},
        uploadStream: {},
        delete: {},
      },
    },
  },
  "content-versioning": {
    enabled: false,
  },
  slugify: {
    enabled: false,
    config: {
      contentTypes: {
        blog: {
          field: "slug",
          references: "title",
        },
      },
    },
  },
  "preview-button": {
    enabled: true,
    resolve: "./src/plugins/preview-button", // path to plugin folder
  },
  "email-designer": {
    enabled: true,

    // ⬇︎ Add the config property
    config: {
      editor: {
        // optional - if you have a premium unlayer account
        // projectId: [UNLAYER_PROJECT_ID],

        tools: {
          heading: {
            properties: {
              text: {
                value: "This is the new default text!",
              },
            },
          },
        },
        options: {
          features: {
            colorPicker: {
              presets: ["#D9E3F0", "#F47373", "#697689", "#37D67A"],
            },
          },
          fonts: {
            showDefaultFonts: false,
            /*
             * If you want use a custom font you need a premium unlayer account and pass a projectId number :-(
             */
            customFonts: [
              {
                label: "Anton",
                value: "'Anton', sans-serif",
                url: "https://fonts.googleapis.com/css?family=Anton",
              },
              {
                label: "Lato",
                value: "'Lato', Tahoma, Verdana, sans-serif",
                url: "https://fonts.googleapis.com/css?family=Lato",
              },
              // ...
            ],
          },
          mergeTags: [
            {
              name: "Naam",
              value: "{{= USER.Naam }}",
              sample: "Naam",
            },
            {
              name: "Voornaam",
              value: "{{= USER.Voornaam }}",
              sample: "Voornaam",
            },
            {
              name: "Email",
              value: "{{= USER.Email }}",
              sample: "john@doe.com",
            },
            {
              name: "Telefoonnummer",
              value: "{{= USER.Telefoonnummer }}",
              sample: "0123456789",
            },
            {
              name: "select",
              value: "{{= USER.select }}",
              sample: "select",
            },
            {
              name: "textArea",
              value: "{{= USER.textArea }}",
              sample: "textArea",
            },
            {
              name: "Form ID",
              value: "{{= USER.formId }}",
              sample: "0",
            },
          ],
        },
        appearance: {
          theme: "dark",
          panels: {
            tools: {
              dock: "left",
            },
          },
        },
      },
    },
  },
  seo: {
    enabled: true,
  },
  ckeditor: {
    enabled: true,
    config: {
      plugin: {
        // disable data-theme tag setting //
        // setAttribute:false,
        // disable strapi theme, will use default ckeditor theme //
        // strapiTheme:false,
        // styles applied to editor container (global scope) //
        // styles:`
        // :root{
        //   --ck-color-focus-border:red;
        //   --ck-color-text:red;
        // }
        // `
        styles: `
        .ck-editor__main p,
        .ck.ck-list__item .ck-button .ck-button__label {
          line-height: normal;
        }`,
      },
      editor: {
        // editor default config

        // https://ckeditor.com/docs/ckeditor5/latest/features/markdown.html
        // if you need markdown support and output set: removePlugins: [''],
        // default is
        removePlugins: ["Markdown"],

        // https://ckeditor.com/docs/ckeditor5/latest/features/toolbar/toolbar.html
        toolbar: {
          items: [
            "paragraph",
            "heading1",
            "heading2",
            "heading3",
            "|",
            "bold",
            "italic",
            "underline",
            "fontSize",
            "removeFormat",
            "|",
            "bulletedList",
            "numberedList",
            "alignment",
            "|",
            "StrapiMediaLib",
            "link",
            "horizontalLine",
            "|",
            "sourceEditing",
            "fullScreen",
            "undo",
            "redo",
          ],
        },
        // https://ckeditor.com/docs/ckeditor5/latest/features/font.html
        fontSize: {
          options: [
            9,
            11,
            13,
            {
              title: "default",
              model: "18px",
            },
            17,
            19,
            21,
            27,
            35,
          ],
          supportAllValues: false,
        },
        fontFamily: {
          options: [
            "default",
            "Arial, Helvetica Neue, Helvetica, Source Sans Pro, sans-serif",
            "Courier New, Courier, monospace",
            "Georgia, serif",
            "Lucida Sans Unicode, Lucida Grande, sans-serif",
            "Tahoma, Geneva, sans-serif",
            "Times New Roman, Times, serif",
            "Trebuchet MS, Helvetica, sans-serif",
            "Verdana, Geneva, sans-serif",
            "Roboto, Roboto Black, Roboto Medium, Roboto Light, sans-serif",
          ],
          supportAllValues: true,
        },
        fontColor: {
          columns: 5,
          documentColors: 10,
        },
        fontBackgroundColor: {
          columns: 5,
          documentColors: 10,
        },
        // https://ckeditor.com/docs/ckeditor5/latest/features/ui-language.html
        // default language: 'en',
        // https://ckeditor.com/docs/ckeditor5/latest/features/images/images-overview.html
        image: {
          resizeUnit: "%",
          resizeOptions: [
            {
              name: "resizeImage:original",
              value: null,
              icon: "original",
            },
            {
              name: "resizeImage:25",
              value: "25",
              icon: "small",
            },
            {
              name: "resizeImage:50",
              value: "50",
              icon: "medium",
            },
            {
              name: "resizeImage:75",
              value: "75",
              icon: "large",
            },
          ],
          toolbar: [
            "toggleImageCaption",
            "imageTextAlternative",
            "imageStyle:inline",
            "imageStyle:block",
            "imageStyle:side",
            "linkImage",
            "resizeImage:25",
            "resizeImage:50",
            "resizeImage:75",
            "resizeImage:original",
          ],
        },
        // https://ckeditor.com/docs/ckeditor5/latest/features/table.html
        table: {
          contentToolbar: [
            "tableColumn",
            "tableRow",
            "mergeTableCells",
            "tableCellProperties",
            "tableProperties",
            "toggleTableCaption",
          ],
        },
        // https://ckeditor.com/docs/ckeditor5/latest/features/headings.html
        heading: {
          options: [
            {
              model: "paragraph",
              title: "Paragraph",
              class: "ck-heading_paragraph",
            },
            {
              model: "heading1",
              view: "h1",
              title: "Heading 1",
              class: "ck-heading_heading1",
            },
            {
              model: "heading2",
              view: "h2",
              title: "Heading 2",
              class: "ck-heading_heading2",
            },
            {
              model: "heading3",
              view: "h3",
              title: "Heading 3",
              class: "ck-heading_heading3",
            },
          ],
        },
        // https://ckeditor.com/docs/ckeditor5/latest/features/general-html-support.html
        htmlSupport: {
          allow: [
            {
              name: "img",
              attributes: {
                sizes: true,
                loading: true,
              },
            },
          ],
        },
      },
    },
  },
});
