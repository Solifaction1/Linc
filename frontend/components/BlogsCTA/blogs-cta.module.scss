@import 'styles/vars';
@import 'styles/utilities';

.blogsCTA {
    @extend %page-container;
    grid-template-areas: 'blogsCTAContainer blogsCTAContainer blogsCTAContainer blogsCTAContainer blogsCTAContainer blogs blogs blogs blogs blogs blogs blogs';

    width: 100%;
    margin: 0;
    padding: 0;

    background-color: $color-red;

    @media (max-width: $break-lg) {
        display: flex;
        flex-direction: column;
    }

    h2, p, a {
        color: $color-red-dark;
    }

    .blogsCTAContainer {
        display: flex;
        flex-direction: column;
        grid-area: blogsCTAContainer;
        gap: calc(#{$spacing-base} * 5);

        padding: calc(#{$spacing-base} * 12.5) calc(#{$spacing-base} * 12) calc(#{$spacing-base} * 12.5) calc(#{$spacing-base} * 18.5);

        @media (max-width: $break-lg) {
            gap: calc(#{$spacing-base} * 4);
            padding: calc(#{$spacing-base} * 7.5) calc(#{$spacing-base} * 3);

            .title {
                font-size: $font-size-title-md;
                line-height: 3,5rem;
            }
        }

        p {
            font-size: $font-size-text-xlg;
            line-height: 3,15rem;

            @media (max-width: $break-lg) {
                font-size: $font-size-text-xs;
                line-height: 2,45rem;
            }
        }

        .cta {
            button {
                border-color: $color-red-dark;
            }

            p {
                color: $color-red-dark;
                line-height: 2,7rem;

                @media (max-width: $break-lg) {
                    font-size: $font-size-text-xlg;
                    line-height: 2,7rem;
                }
            }
        }
    }

    .blogs {
        grid-area: blogs;
        padding: calc(#{$spacing-base} * 12.5) calc(#{$spacing-base} * 18.5) calc(#{$spacing-base} * 12.5) 0;
    
        @media (max-width: $break-lg) {
            padding: 0 calc(#{$spacing-base} * 3) calc(#{$spacing-base} * 7.5) calc(#{$spacing-base} * 3);
        }
        
        .cta {
            @extend %accordion-link;

            border-bottom: 1px solid $color-red-dark;

            &:first-of-type {
                border-top: 1px solid $color-red-dark;
            }

            svg {
                fill: $color-red-dark;
            }
        }
    }
}