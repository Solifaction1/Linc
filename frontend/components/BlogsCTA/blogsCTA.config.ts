import BlogsCTAData from '@models/blogsCTA.model'

const defaultProps: BlogsCTAData = {
	title: 'title',
	description: 'description',
	cta: {
		id: 0,
		title: 'cta title',
		link: '/link',
		externalLink: false,
		buttonType: 'Button primary',
    },
    collectionType: {
		data: [
			{
				id: 1,
				attributes: {
					title: 'Blog 1',
					slug: 'slug-1',
					image: {
						data: null
					},
					component: [],
					locale: 'locale',
					seo: null,
					createdAt: '',
					updatedAt: '',
					publishedAt: '',
					body: [],
					color: 'gray'
				}
			},
			{
				id: 2,
				attributes: {
					title: 'Blog 2',
					slug: 'slug-2',
					image: {
						data: null
					},
					component: [],
					locale: 'locale',
					seo: null,
					createdAt: '',
					updatedAt: '',
					publishedAt: '',
					body: [],
					color: 'gray'
				}	
			}
		]
	}
}

export { defaultProps }
