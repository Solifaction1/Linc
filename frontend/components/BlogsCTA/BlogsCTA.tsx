import React from 'react'
import BlogsCTAData from '@models/blogsCTA.model'
import SwitchCTA from '@templates/SwitchCTA'
import SwitchIcon from '@templates/SwitchIcon'
import { Icon } from '@templates/switch-icon.config'
import { defaultProps } from './blogsCTA.config'
import styles from './blogs-cta.module.scss'

const BlogsCTA = (props: BlogsCTAData) => (
	<article className={styles.blogsCTA}>
		<div className={styles.blogsCTAContainer}>
			{props.title && <h2 className={styles.title}>{props.title}</h2>}
			{props.description && (
				<p className={styles.description}>{props.description}</p>
			)}
			{props.cta && (
				<div className={styles.cta}>
					<SwitchCTA ctaData={props.cta} />
				</div>
			)}
		</div>

		<div className={styles.blogs}>
			{props.collectionType?.data.map((rechtsDomein) => (
				<div className={styles.cta} key={rechtsDomein.id}>
					<SwitchCTA
						ctaData={{
							title: rechtsDomein.attributes.title ?? '',
							link: '/',
							externalLink: false,
							buttonType: 'Link',
						}}
					/>

					<SwitchIcon icon={Icon.ANGLE_RIGHT} />
				</div>
			))}
		</div>
	</article>
)

BlogsCTA.defaultProps = defaultProps

export default BlogsCTA
