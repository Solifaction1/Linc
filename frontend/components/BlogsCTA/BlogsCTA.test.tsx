import React from 'react'

import { render } from '@testing-library/react'
import '@testing-library/jest-dom'
import { defaultProps } from './blogsCTA.config'
import BlogsCTA from './BlogsCTA'

jest.mock('next/router', () => ({
	useRouter: jest.fn(),
}))

describe('BlogsCTA', () => {
	it('renders a BlogsCTA', () => {
		const { container, getByText } = render(<BlogsCTA {...defaultProps} />)

		expect(container).toMatchSnapshot()

		expect(container.querySelector('article')).toBeVisible()
		expect(container.getElementsByClassName('blogsCTA')[0]).toBeVisible()

		expect(
			container.getElementsByClassName('blogsCTAContainer')[0]
		).toBeVisible()

		expect(container.getElementsByClassName('title')[0]).toBeVisible()
		expect(getByText('title', { selector: 'h2' })).toBeVisible()

		expect(container.getElementsByClassName('description')[0]).toBeVisible()
		expect(getByText('description', { selector: 'p' })).toBeVisible()

		expect(container.getElementsByClassName('cta')[0]).toBeVisible()
		expect(
			container.getElementsByClassName('button false primary md false')[0]
		).toBeVisible()
		expect(container.getElementsByClassName('label')[0]).toBeVisible()
		expect(getByText('cta title', { selector: 'p' })).toBeVisible()

		expect(container.getElementsByClassName('blogs')[0]).toBeVisible()

		expect(container.getElementsByClassName('cta')[1]).toBeVisible()
		expect(
			container.getElementsByClassName('link link md false')[0]
		).toBeVisible()
		expect(container.querySelectorAll('a[href="/"]')[0]).toBeVisible()
		expect(getByText('Blog 1', { selector: 'a' })).toBeVisible()
		expect(container.querySelectorAll('svg')[0]).toBeVisible()
		expect(container.querySelectorAll('path')[0]).toBeVisible()

		expect(container.getElementsByClassName('cta')[2]).toBeVisible()
		expect(
			container.getElementsByClassName('link link md false')[1]
		).toBeVisible()
		expect(container.querySelectorAll('a[href="/"]')[1]).toBeVisible()
		expect(getByText('Blog 2', { selector: 'a' })).toBeVisible()
		expect(container.querySelectorAll('svg')[1]).toBeVisible()
		expect(container.querySelectorAll('path')[1]).toBeVisible()
	})
})
