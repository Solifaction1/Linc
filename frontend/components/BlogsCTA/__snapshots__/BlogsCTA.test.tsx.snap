// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlogsCTA renders a BlogsCTA 1`] = `
<div>
  <article
    class="blogsCTA"
  >
    <div
      class="blogsCTAContainer"
    >
      <h2
        class="title"
      >
        title
      </h2>
      <p
        class="description"
      >
        description
      </p>
      <div
        class="cta"
      >
        <button
          class="button false primary md false  "
          type="button"
        >
          <p
            class="label"
          >
            cta title
          </p>
        </button>
      </div>
    </div>
    <div
      class="blogs"
    >
      <div
        class="cta"
      >
        <a
          class="link  link md false"
          href="/"
        >
          Blog 1
        </a>
        <svg
          fill="currentColor"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M14.83,11.29,10.59,7.05a1,1,0,0,0-1.42,0,1,1,0,0,0,0,1.41L12.71,12,9.17,15.54a1,1,0,0,0,0,1.41,1,1,0,0,0,.71.29,1,1,0,0,0,.71-.29l4.24-4.24A1,1,0,0,0,14.83,11.29Z"
          />
        </svg>
      </div>
      <div
        class="cta"
      >
        <a
          class="link  link md false"
          href="/"
        >
          Blog 2
        </a>
        <svg
          fill="currentColor"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M14.83,11.29,10.59,7.05a1,1,0,0,0-1.42,0,1,1,0,0,0,0,1.41L12.71,12,9.17,15.54a1,1,0,0,0,0,1.41,1,1,0,0,0,.71.29,1,1,0,0,0,.71-.29l4.24-4.24A1,1,0,0,0,14.83,11.29Z"
          />
        </svg>
      </div>
    </div>
  </article>
</div>
`;
